### PR TITLE
Fix recording

### PR DESCRIPTION
**Pull request by StianJohnsen**
Have gotten recording to work.
Saves accelerometer/gyroscope and location data to a .csv file.
Saves information about the data in a .json file.
Checkboxes work for configuring the recording.
Recording control (pause/resume, stop and delete buttons) also work

### CODE REVIEW

File: app/src/main/res/layout/fragment_recording.xml
Comment (ysz0507) [inline app/src/main/res/layout/fragment_recording.xml:1]: We have to talk about the design tomorrow. I am not sure what out target is

Comment (ysz0507) [inline app/src/main/res/layout/fragment_recording.xml:1]: There are a few Bugs:
- back button does not work
- crash though going backwards
- the checkboxes do not work

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt:1]: Really important: make a rebase manually on your computer an fix the errors that will occur.

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt [hunk 1]
```diff
@@ -45,15 +51,45 @@ class ConfigureRecordingFragment : BaseFragment<FragmentConfigureRecordingBindin
...
+        args.putBoolean("isFiltered", isFilteredChecked)
+        args.putBoolean("isAccel", isAccelChecked)
+        args.putBoolean("isGyro", isGyroChecked)
+        /*
+                val action = ConfigureRecordingFragmentDirections.actionActionConfigureToRecordingFragment(
+                    isFilteredChecked,
+                    isAccelChecked,
+                    isGyroChecked
+                )
+         */
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt:90]: remove commented code

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt [hunk 2]
```diff
@@ -17,9 +18,14 @@ class ConfigureRecordingFragment : BaseFragment<FragmentConfigureRecordingBindin
 ) {
     private lateinit var bottomNavigationView: BottomNavigationView
     private val viewModel: ConfigureRecordingViewModel by viewModels()
+    private val settingsViewModel: SettingsViewModel by viewModels()
 
     private var startTimeMillis: Long = 0
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt:23]: you are not using this variable

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt [hunk 3]
```diff
@@ -17,9 +18,14 @@ class ConfigureRecordingFragment : BaseFragment<FragmentConfigureRecordingBindin
...
 
+    var isFilteredChecked = false
+    var isAccelChecked = false
+    var isGyroChecked = false
+
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt:31]: Your IDE is highlighting some stuff yellow. It is most of the time very useful to implement it to get cleaner code. Here as well you don't have to check for null twice

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt [hunk 4]
```diff
@@ -45,15 +51,45 @@ class ConfigureRecordingFragment : BaseFragment<FragmentConfigureRecordingBindin
...
-        }
-        binding.imageBackConfigure.setOnClickListener {
-            findNavController().navigate(R.id.action_action_configure_to_SettingsFragment)
-        }
+
+
     }
 
+
     fun startRecording() {
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/ConfigureRecordingFragment.kt:78]: Yellow again (private)
Just fix everything that's yellow

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt [hunk 5]
```diff
@@ -1,36 +1,87 @@
...
+import java.nio.charset.Charset
+import java.time.LocalDateTime
 
 class RecordingFragment : BaseFragment<FragmentRecordingBinding>(
     FragmentRecordingBinding::inflate
-), SensorEventListener {
+), SensorEventListener, LocationListener {
     private lateinit var bottomNavigationView: BottomNavigationView
-    private val viewModel: RecordingViewModel by viewModels()
+    //private val viewModel: RecordingViewModel by viewModels()
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt:38]: remove comment

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt [hunk 6]
```diff
@@ -83,6 +145,17 @@ class RecordingFragment : BaseFragment<FragmentRecordingBinding>(
...
+
+        /*
+                settingsViewModel.isAccelerometerChecked.observe(this.viewLifecycleOwner) {
+            Log.d("observer", "accel: " + it.toString())
+        }
+
+        settingsViewModel.isGyroChecked.observe(this.viewLifecycleOwner) {
+            Log.d("observer", "gyro: " + it.toString())
+        }
+         */
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt:157]: remove

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt [hunk 7]
```diff
@@ -94,8 +167,19 @@ class RecordingFragment : BaseFragment<FragmentRecordingBinding>(
         TODO("Not yet implemented")
     }
 
+    override fun onDestroyView() {
+        super.onDestroyView()
+        // Cancel the coroutine job associated with the collection
+        //settingsViewModel.isFilteredChecked.collect {}.cancel()
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt:173]: remove function override

File: app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt [hunk 8]
```diff
@@ -68,6 +126,10 @@ class RecordingFragment : BaseFragment<FragmentRecordingBinding>(
 
 
     private var startTimeMillis: Long = 0
+    private var pauseTimeMillis: Long = 0
```
Comment (ysz0507) [inline app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingFragment.kt:129]: never used

**Review by ysz0507 — CHANGES_REQUESTED** [review]
I think that is enough for this review. We have to talk about what has to be done tomorrow in detail. When you are done I will give you another review. So far I like your ideas 👍

**Review by QuentinRG — APPROVED** [review]
Well done! nothing to say, no warnings and it's working

**Review by ysz0507 — APPROVED** [review]
nice

**Pull request merged**